addons/binding/org.openhab.binding.allplay/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,177 @@
+		<item-type>Player</item-type>
+		<label>Control</label>
+		<description>Control the AllPlay speaker, e.g. start/pause/next/previous/ffward/rewind</description>
+		<category>Player</category>
+	</channel-type>
+	<channel-type id="coverart">
+		<item-type>Image</item-type>
+		<label>Cover Art</label>
+		<description>Cover art image of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="coverarturl" advanced="true">
+		<item-type>String</item-type>
+		<label>Cover Art URL</label>
+		<description>Cover art URL of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentalbum">
+		<item-type>String</item-type>
+		<label>Current Album</label>
+		<description>Album of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentartist">
+		<item-type>String</item-type>
+		<label>Current Artist</label>
+		<description>Artist of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentduration" advanced="true">
+		<item-type>Number</item-type>
+		<label>Current Duration</label>
+		<description>Duration in milliseconds of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentgenre" advanced="true">
+		<item-type>String</item-type>
+		<label>Current Genre</label>
+		<description>Genre of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currenttitle">
+		<item-type>String</item-type>
+		<label>Current Title</label>
+		<description>Title of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentuserdata" advanced="true">
+		<item-type>String</item-type>
+		<label>Current User Data</label>
+		<description>Custom user data (e.g. name of radio station) of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currenturl" advanced="true">
+		<item-type>String</item-type>
+		<label>Current URL</label>
+		<description>URL of the track or radio station currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="loopmode" advanced="true">
+		<item-type>String</item-type>
+		<label>Loop Mode</label>
+		<description>Loop mode of the speaker (ONE, ALL, NONE)</description>
+		<state>
+			<options>
+				<option value="NONE">No Repeat</option>
+				<option value="ONE">Repeat Track</option>
+				<option value="ALL">Repeat Playlist</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="mute" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Mute</label>
+		<description>Set or get the mute state of the speaker</description>
+	</channel-type>
+	<channel-type id="playstate">
```

Comment:
At least I am using it (in my rules) to change the volume before the speaker starts playing an item. If the speaker is in standby, I cannot change the volume (I really believe the Panasonic speakers have some design flaws), but I can start playing an item. So, on a quiet Sunday morning, I poll the playstate and change the volume as soon as I know the speaker has woken up from standby and before the speaker actually starts playing with the volume of Satuday evenings movie night 😉 .